controllers/grid/users/reviewer/form/AdvancedSearchReviewerForm.php

```diff
@@ -144,6 +144,21 @@ public function fetch($request, $template = null, $display = false)
         $warnOnAssignment = array_merge($warnOnAssignment, $userIds);
         $warnOnAssignment = array_values(array_unique(array_map('intval', $warnOnAssignment)));
 
+        $locale = Locale::getLocale();
+        $submissionAuthors = $this->getSubmission()->getCurrentPublication()->getData('authors');
+        $authorAffiliations = [];
+        $authors = [];
+        foreach($submissionAuthors as $submissionAuthor) {
+            $affiliations = [];
+            foreach($submissionAuthor->getData('affiliation') as $affiliation) {
+                if(!empty($affiliation)) {
+                    $authorAffiliations[] = $affiliation;
+                    $affiliations[] = $affiliation;
+                }
+            }
+            $authors[$submissionAuthor->getFullName(true, false, $locale)] = implode(',', array_filter($affiliations));
```

Comment:
Affiliations in different locales are concatenated here as a string. Will this work with [affiliation match](https://github.com/pkp/ui-library/blob/9a4dcffc39323ca56fd1ee279faab7287b37a0ca/src/components/ListPanel/users/SelectReviewerListItem.vue#L462) feature. I mean if the match is only for one locale, would it show the match label anyway? If not, we can pass it as an array and concatenate on the front-end after comparison. Or just leave only affiliation in the same locale as in the reviewers list.

Reply:
And I was mistaken about one of the previous recommendations. Looking at the `Identity::getFullName()`, it will have the same effect without additional arguments as you initially coded. It uses request locale by default, not the submission locale, and the Author class doesn't modify this behaviour.

Reply:
Very good spot! So I have just changed affiliation retrieval to just show one given the locale like this 
        $authorAffiliations = [];
        $authors = [];
        foreach($submissionAuthors as $submissionAuthor) {
            $affiliation = $submissionAuthor->getData('affiliation')[$locale];
            $authorAffiliations[] = $affiliation;
            $authors[$submissionAuthor->getFullName(true, false, $locale)] = $affiliation;
        }`

There is only one shown to match with in `SelectReviewerListItem` anyway as shown here `{{ localize(item.affiliation) }}` so this works

Reply:
Thanks! Added a quick comment